runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,449 @@
+  {cmd}` on a command defined as a Lua callback. Similarly to keymaps, it is
+  recommended to add a `desc` key to commands defined as Lua functions.
+• `force` is equivalent to calling `:command!` and replaces a command if one
+  with the same name already exists. It is true by default, unlike its
+  Vim script equivalent.
+
+The `-complete` attribute can take a Lua function in addition to the
+attributes listed in |:command-complete|. >lua
+
+	vim.api.nvim_create_user_command('Upper', function() end, {
+	    nargs = 1,
+	    complete = function(ArgLead, CmdLine, CursorPos)
+		-- return completion candidates as a list-like table
+		return { 'foo', 'bar', 'baz' }
+	    end,
+	})
+
+Buffer-local user commands also take a buffer number as their first argument.
+This is an advantage over `-buffer` which can only define a command for the
+current buffer. >lua
+
+	vim.api.nvim_buf_create_user_command(4, 'Upper', function() end, {})
+
+
+-------------------------------------------------------------------------------
+Deleting user commands
+TODO: write
+
+vim.api.|nvim_del_user_command()| takes a command name. >lua
+
+	vim.api.nvim_del_user_command('Upper')
+	-- :delcommand Upper
+
+Again, vim.api.|nvim_buf_del_user_command()|, takes a buffer number as its
+first argument, with `0` representing the current buffer. >lua
+
+	vim.api.nvim_buf_del_user_command(4, 'Upper')
+
+See also:
+- |nvim_create_user_command()
+- |40.2|
+- |command-attributes|
+
+-------------------------------------------------------------------------------
+Caveats
+
+The `-complete=custom` attribute automatically filters completion candidates
+and has built-in wildcard (|wildcard|) support: >vim
+
+	function! s:completion_function(ArgLead, CmdLine, CursorPos) abort
+	    return join([
+		\ 'strawberry',
+		\ 'star',
+		\ 'stellar',
+		\ ], "\n")
+	endfunction
+
+	command! -nargs=1 -complete=custom,s:completion_function Test echo <q-args>
+	" Typing `:Test st[ae]<Tab>` returns "star" and "stellar"
+
+Passing a Lua function to `complete` makes it behave like `customlist` which
+leaves filtering up to the user: >lua
+
+	vim.api.nvim_create_user_command('Test', function() end, {
+	    nargs = 1,
+	    complete = function(ArgLead, CmdLine, CursorPos)
+		return {
+		    'strawberry',
+		    'star',
+		    'stellar',
+		}
+	    end,
+	})
+
+	-- Typing `:Test z<Tab>` returns all the completion results because the list was not filtered
+
+==============================================================================
+Further reading:
+TODO: write
+• vim.loop
+• vim.lsp
+• vim.treesitter
+
+==============================================================================
+
+TODO: maybe add a "General tips and recommendations" section like nanotee
+does? Depends on how long this will get.
+
+This guide is in large part taken from nanotee's Lua guide:
+https://github.com/nanotee/nvim-lua-guide
+
+Thank you nanotee :)
+
+vim:tw=78:ts=8:noet:ft=help:norl:
```

Comment:
I'd set the indentation to only spaces instead of mixing tabs and spaces
```suggestion
vim:tw=78:sts=4:sw=4:et:ft=help:norl:
```

Reply:
"Neovim" help files tend to have
```vim
 vim:tw=78:ts=8:sw=4:sts=4:et:ft=help:norl:
```

Reply:
And please, no mixing of tabs and spaces!